storage/Devices/LvmVgImpl.cc

```diff
@@ -520,4 +520,102 @@ namespace storage
 	return lhs->get_vg_name() < rhs->get_vg_name();
     }
 
+
+    void
+    LvmVg::Impl::add_dependencies(Actiongraph::Impl& actiongraph) const
+    {
+	// First, all the operations removing or shrinking LVs
+	vector<Actiongraph::Impl::vertex_descriptor> decrease_lv_actions;
+	// Then a sync point
+	Action::Base* decrease_lv_sync = new Action::Create(get_sid(), true);
```

Comment:
In the beginning there was an Action::Nop but it also had problems. AFAIR even for Nop you need the information whether it's create or delete so that you can e.g. delete children before the parents - even if the delete action itself is a nop.